File: solutions/typescript/scrabble-score/2/scrabble-score.ts
const LETTER_SCORES: { [key: string]: number } = {};

const LETTER_GROUPS: [string, number][] = [
  ['AEIOULNRST', 1],
  ['DG', 2],
  ['BCMP', 3],
  ['FHVWY', 4],
  ['K', 5],
  ['JX', 8],
  ['QZ', 10],
];

LETTER_GROUPS.forEach(([group, score]: [string, number]): void =>
  [...group].forEach((letter: string) => (LETTER_SCORES[letter] = score))
);

export function score(word: string | undefined): number {
  if (!word) return 0;
  return [...word.toUpperCase()].reduce(
    (total: number, letter: string) => total + LETTER_SCORES[letter],
    0
  );
}

File: solutions/python/scrabble-score/2/scrabble_score.py
def score(word):
    LETTERS = list("AEIOULNRSTDGBCMPFHVWYKJXQZ")
    VALUES = [int(digit) + 1 for digit in "00000000001122223333347799"]

    LETTER_SCORES = {k: v for k, v in zip(LETTERS, VALUES)}
        
    return sum([LETTER_SCORES[char] for char in word.upper()])
